pkgs/by-name/hy/hyprwall/package.nix

```
@@ -0,0 +1,68 @@
+{
+  lib,
+  fetchurl,
+  fetchFromGitHub,
+  rustPlatform,
+
+  pkg-config,
+  glib,
+  gdk-pixbuf,
+  pango,
+  graphene,
+  gtk4,
+  gtk3,
+  git,
+}:
+
+rustPlatform.buildRustPackage rec {
+  pname = "hyprwall";
+  version = "0.1.8";
+
+  src = fetchFromGitHub {
+    owner = "hyprutils";
+    repo = "hyprwall";
+    rev = "refs/tags/v${version}";
+    hash = "sha256-dzPd+5cws3hKhdd1CKKEO7EWMS0XW0y1vqxg1XKX+gY=";
+  };
+
+  cargoHash = "sha256-gT2ysWHckcUl1yx5tciy6kSvZZ0srrs4OwI1mr/58Pc=";
+
+  nativeBuildInputs = [
+    pkg-config
+    glib
+    gdk-pixbuf
+    pango
+    graphene
+    gtk4
+    gtk3
+    git
+  ];
+
+  buildInputs = [
+    pkg-config
+    glib
+    gdk-pixbuf
+    pango
+    graphene
+    gtk4
+    gtk3
+  ];
+
+  postInstall = ''
+    mkdir -p "$out/share/applications"
+    mkdir -p "$out/share/icons"
+    cp hyprwall.desktop "$out/share/applications"
+    cp hyprwall.png "$out/share/icons"
+    substituteInPlace "$out/share/applications/hyprwall.desktop" --replace 'Exec=/usr/bin/hyprwall' "Exec=$out/bin/hyprwall"
```

```suggestion
    substituteInPlace "$out/share/applications/hyprwall.desktop" \
      --replace 'Exec=/usr/bin/hyprwall' "Exec=hyprwall"
```

https://github.com/NixOS/nixpkgs/issues/308324

---

Aside from this code issue,do you have any links to documentation on how to use substituteInPlace, I wanted to build a python package derivation but the imports check fails.

---

https://nixos.org/manual/nixpkgs/unstable/#fun-substituteInPlace

---

This is not resolved